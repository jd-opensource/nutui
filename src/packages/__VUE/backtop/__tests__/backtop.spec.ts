import { mount, config } from '@vue/test-utils';
import BackTop from '../index.vue';

import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('emit click event', () => {
  const wrapper = mount(BackTop);
  wrapper.trigger('click');
  expect(wrapper.emitted('click')!.length).toEqual(1);
});

test('backtop show', () => {
  const wrapper = mount(BackTop);
  const backtop = wrapper.find('.nut-backtop');
  expect(backtop.exists()).toBe(true);
});

test('backtop style', () => {
  const wrapper = mount(BackTop, {
    props: {
      bottom: 50
    }
  });
  const backtop: any = wrapper.find('.nut-backtop');
  expect(backtop.element.style.bottom).toBe('50px');
});

test('backtop style', () => {
  const wrapper = mount(BackTop, {
    props: {
      right: 40
    }
  });
  const backtop: any = wrapper.find('.nut-backtop');
  expect(backtop.element.style.right).toBe('40px');
});

test('backtop style', () => {
  const wrapper = mount(BackTop, {
    props: {
      zIndex: 100
    }
  });
  const backtop: any = wrapper.find('.nut-backtop');
  expect(backtop.element.style.zIndex).toBe('100');
});

test('should allow to use the elId prop', () => {
  const root = document.createElement('div');
  mount(BackTop, {
    props: {
      elId: root
    }
  });
  expect(root.innerHTML).toMatchSnapshot();
});
