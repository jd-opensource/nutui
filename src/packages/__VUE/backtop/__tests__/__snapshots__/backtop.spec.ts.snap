// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow to use the elId prop 1`] = `""`;

exports[`should allow to use the teleport prop 1`] = `""`;
